### PR TITLE
feature/io-421_base_architecture

### PR DESCRIPTION
**Pull request by egor-slatvinskiy-idf**

### CODE REVIEW

File: packages/data/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: data
+description: A new Flutter package project.
+version: 0.0.1
+
```
Comment (vladimir-hodaryonok-idf) [inline packages/data/pubspec.yaml:5]: Publishable packages can't have 'path' dependencies.
add - publish_to: 'none'
and check other pub specs

File: packages/presentation/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+dev_dependencies:
+  flutter_test:
+    sdk: flutter
+  flutter_lints: ^2.0.0
+
+flutter:
+   assets:
+     - images/
+  # fonts:
```
Comment (vladimir-hodaryonok-idf) [inline packages/presentation/pubspec.yaml:23]: comments

File: pubspec.lock [hunk 3]
```diff
@@ -50,6 +50,20 @@ packages:
       url: "https://pub.dartlang.org"
```
Comment (vladimir-hodaryonok-idf) [inline pubspec.lock:50]: pls^ add this file to .gitignore

**Review by vladimir-hodaryonok-idf — APPROVED** [review]
on my opinion better to import svg files instead png or jpg. this is vector format and will look great on every device. But up to u

**Review by vladimir-hodaryonok-idf — APPROVED** [review]

**Review by elizaveta-seleznyova-idf — APPROVED** [review]

File: packages/presentation/pubspec.yaml (hunk 2 above)
Comment (egor-slatvinskiy-idf) [reply packages/presentation/pubspec.yaml:23]: for the future

File: packages/presentation/lib/base/bloc_screen.dart [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import 'package:flutter/material.dart';
+import 'package:get_it/get_it.dart';
+import 'package:presentation/base/bloc.dart';
+
+abstract class BlocScreenState<BS extends StatefulWidget, B extends Bloc>
+    extends State<BS> with WidgetsBindingObserver {
```
Comment (alexei-kruk-idf) [inline packages/presentation/lib/base/bloc_screen.dart]: i think, that you won't use WidgetsBindingObserver, so you can remove WidgetsBindingObserver

File: packages/presentation/lib/my_app.dart [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+class _MyAppState extends BlocScreenState<StatefulWidget, AppBloc> {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      color: const Color.fromRGBO(15, 27, 43, 1),
+      theme: ThemeData(
+        canvasColor: const Color.fromRGBO(15, 27, 43, 1),
+        appBarTheme: const AppBarTheme(
+          backgroundColor: Color.fromRGBO(15, 27, 43, 1),
```
Comment (alexei-kruk-idf) [inline packages/presentation/lib/my_app.dart]: create abstract or singleton class AppColors and get colors from it

File: packages/presentation/lib/ui/splash_screen/splash_widget.dart [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      body: Center(
+        child: Container(
+          decoration: const BoxDecoration(
+            gradient: LinearGradient(
+              begin: Alignment.topCenter,
+              end: Alignment.bottomCenter,
+              colors: [
+                Color.fromRGBO(229, 25, 55, 1),
+                Color.fromRGBO(219, 82, 82, 1),
+              ],
```
Comment (alexei-kruk-idf) [inline packages/presentation/lib/ui/splash_screen/splash_widget.dart]: -//-

File: packages/presentation/lib/ui/splash_screen/splash_widget.dart [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+            ),
+          ),
+          child: Center(
+            child: Image.asset(
+              image,
+              // fit: BoxFit.cover,
```
Comment (alexei-kruk-idf) [inline packages/presentation/lib/ui/splash_screen/splash_widget.dart]: remove

**Review by alexei-kruk-idf — APPROVED** [review]

File: packages/presentation/lib/ui/home_page/home_widget.dart [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+class _HomeWidgetState extends State<HomeWidget> {
+  int _selectedTab = 0;
+
+  void onSelectedTab(int index) {
+    if (_selectedTab == index) return;
+    setState(
+      () {
+        _selectedTab = index;
+      },
+    );
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/ui/home_page/home_widget.dart]: now it's ok, but later it should be moved to bloc

File: packages/presentation/lib/ui/home_page/home_widget.dart [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: const Color.fromRGBO(15, 27, 43, 1),
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/ui/home_page/home_widget.dart]: move to file with constants

File: packages/presentation/lib/ui/movie_page/movie_now_showing_widget.dart [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+              children: const [
+                Text(
+                  'Title',
+                  style: TextStyle(color: Colors.white),
+                  textAlign: TextAlign.start,
+                ),
+                Icon(
+                  Icons.star,
+                  size: 13,
+                  color: Color.fromRGBO(255, 192, 69, 1),
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/ui/movie_page/movie_now_showing_widget.dart]: -//-

File: packages/presentation/lib/ui/movie_page/movie_widget.dart [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+
+class MovieWidget extends StatelessWidget {
+  const MovieWidget({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return DefaultTabController(
+      length: 2,
+      child: Scaffold(
+        backgroundColor: const Color.fromRGBO(15, 27, 43, 1),
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/ui/movie_page/movie_widget.dart]: -//-

File: packages/presentation/lib/ui/movie_page/movie_widget.dart [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+              margin: const EdgeInsets.only(
+                bottom: 24,
+                top: 7,
+                right: 17,
+                left: 18,
+              ),
+              height: 50,
+              decoration: BoxDecoration(
+                border: Border.all(color: const Color.fromRGBO(44, 63, 91, 1)),
+                color: const Color.fromRGBO(15, 27, 43, 1),
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/ui/movie_page/movie_widget.dart]: -//-

File: packages/presentation/lib/ui/splash_screen/bloc/splash_bloc.dart [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+  }
+
+  Future<void> nextScreen() async {
+    await Future.delayed(
+      const Duration(seconds: 3),
+      () => appNavigator.popAndPush(
+        HomeWidget.page(),
+      ),
+    );
+  }
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/ui/splash_screen/bloc/splash_bloc.dart]: move delay to usecase

File: packages/presentation/lib/ui/splash_screen/splash_widget.dart [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+        return Scaffold(
+          body: Center(
+            child: Container(
+              decoration: const BoxDecoration(
+                gradient: LinearGradient(
+                  begin: Alignment.topCenter,
+                  end: Alignment.bottomCenter,
+                  colors: [
+                    Color.fromRGBO(229, 25, 55, 1),
+                    Color.fromRGBO(219, 82, 82, 1),
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/ui/splash_screen/splash_widget.dart]: -//-

File: pubspec.lock (hunk 3 above)
Comment (sergei-mikhailovskii-idf) [reply pubspec.lock:50]: and delete the file

File: pubspec.yaml [hunk 15]
```diff
@@ -9,6 +9,10 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
+  presentation:
+    path: packages/presentation
+  data:
+    path: packages/data
```
Comment (sergei-mikhailovskii-idf) [inline pubspec.yaml:15]: presentation already contains it

**Review by sergei-mikhailovskii-idf — APPROVED** [review]
if it is possible, remove splash and main screen logic from this branch

**Pull request merged**